### PR TITLE
Set notification time on file upload and download

### LINKED ISSUE #2188: Notification: Wrong timestamp is shown
### Actual behaviour
screenshot_20180414-214804

### Expected behaviour
The timestamp should be the time I started the upload (~2 seconds before)
 
### Steps to reproduce
1. Open app
2. Upload image

### Environment data
Android version: 7.1

Device model: Oneplus X

Stock or customized system: LineageOS 14

ownCloud app version: 2.7.0-beta1

Just from looking at the code, there is a "setWhen" missing: https://github.com/owncloud/android/blob/master/src/com/owncloud/android/files/services/FileUploader.java#L900
Same for FileDownloader.java and maybe more

### PR DESCRIPTION
**Pull request by mueller-ma**
Android Studio couldn't build an apk with the gradle version so I had to
update.

Fixes #2188

Signed-off-by: mueller-ma <mueller-ma@users.noreply.github.com>

### CODE REVIEW

**Comment by davigonz** [comment]
Hi @mueller-ma , glad to see you have finally opened a PR to fix #2188 , thank you so much.

Let me explain you the next steps we follow after opening a PR. First, I will have a look at the code and request some changes if any. After that, we will perform some tests and if everything has gone fine, your contribution will be merged and included in coming versions.

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,7 @@
-#Tue Apr 18 18:43:33 CEST 2017
+#Wed Apr 18 16:00:24 CEST 2018
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-3.3-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.4-all.zip
+distributionSha256Sum=7a2c66d1a78f811d5f37d14630ad21cec5e77a2a4dc61e787e2257a6341016ce
```
Comment (davigonz) [inline gradle/wrapper/gradle-wrapper.properties]: What is this line used for?

Comment (mueller-ma) [reply gradle/wrapper/gradle-wrapper.properties]: The gradle wrapper downloads the gradle binary (https\://services.gradle.org/distributions/gradle-4.4-all.zip) and can verify if the binary matches the hash sum. It adds a bit more security, not much, but why not.

**Comment by davigonz** [comment]
Hi again @mueller-ma , let's move this contribution forward, can you fix the conflicting files? Thanks

**Comment by davigonz** [comment]
@mueller-ma we usually work with rebase instead of merge and I see two merges from master into your branch. Have a look at https://github.com/owncloud/android/pull/2075#issuecomment-359428847 for more details to solve it.

**Comment by mueller-ma** [comment]
Just click on "Squash and merge" and the commits will be squashed by GitHub :)

**Comment by davigonz** [comment]
Hi @mueller-ma, thanks for the suggestion but not sure if that would work for us. 

It's not a matter of number of commits but history, the `2188-time` commits with name `Merge branch 'master' into 2188-time` make all master commits appear in the `2188-time` commits history (`git log`) and your `Set notification time on file upload and download` commit will appear between some of those commits so if we squash them think we will have just one commit that will contain the master history with `Set notification time on file upload and download` in the middle and we are planning to merge that into master when we just want `Set notification time on file upload and download` commit placed after the last commit of the master branch, what would be possible if we use `git rebase`.

**Comment by davigonz** [comment]
Hi again @mueller-ma , can you please rebase this again? I will merge it once the branch is updated with the latest changes in master, thanks

**Comment by jesmrec** [comment]
Bug fixed. Approved from my side.

**Comment by davigonz** [comment]
> Bug fixed. Approved from my side.

Great, thanks, this will be merged via https://github.com/owncloud/android/pull/2250

**Pull request closed without merging**